main.go

```diff
@@ -88,10 +91,29 @@ func main() {
 		klog.Fatal("failed to configure k8s client")
 	}
 
+	// secret manager client, without auth so that authentication can be
+	// re-added on a per-RPC basis for each mount
+	smOpts := []option.ClientOption{
+		option.WithUserAgent(ua),
+		// tell the secretmanager library to not add transport-level ADC since we need to override on a per call basis
+		option.WithoutAuthentication(),
+		// grpc oauth TokenSource credentials require transport security, so this must be set explicitly even though TLS
+		// is used
+		option.WithGRPCDialOption(grpc.WithTransportCredentials(credentials.NewTLS(nil))),
+		option.WithGRPCConnectionPool(5),
+	}
+
+	// Build the secret manager client
+	sc, err := secretmanager.NewClient(ctx, smOpts...)
+	if err != nil {
+		klog.ErrorS(err, "failed to create secretmanager client")
+		klog.Fatal("failed to create secretmanager client")
```

Comment:
Why not just: klog.Fatalf("failed to create secretmanager client %v", err)?

Reply:
`ErrorS` nicely formats `"err":"<msg>"` key/values in the log line so you can filter/search logs better. `Fatalf` does not provide access to additional key/value message fields so `"err"` would not be filter-able.